pkg/controller/atlasdatabaseuser/atlasdatabaseuser_controller.go

```diff
@@ -123,18 +124,16 @@ func (r *AtlasDatabaseUserReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		return result.ReconcileResult(), nil
 	}
 
-	atlasClient, orgID, err := r.AtlasProvider.Client(ctx, project.ConnectionSecretObjectKey(), log)
+	dus, err := dbuser.NewAtlasDatabaseUsersService(ctx, r.AtlasProvider, project.ConnectionSecretObjectKey(), log)
 	if err != nil {
 		result = workflow.Terminate(workflow.AtlasAPIAccessNotConfigured, err.Error())
 		workflowCtx.SetConditionFromResult(api.DatabaseUserReadyType, result)
 
 		return result.ReconcileResult(), nil
 	}
-	workflowCtx.OrgID = orgID
-	workflowCtx.Client = atlasClient
 
 	// Setting protection flag to static false because ownership detection is disabled.
-	owner, err := customresource.IsOwner(databaseUser, false, customresource.IsResourceManagedByOperator, managedByAtlas(ctx, atlasClient, project.ID(), log))
+	owner, err := customresource.IsOwner(databaseUser, false, customresource.IsResourceManagedByOperator, managedByAtlas(ctx, dus, project.ID(), log))
```

Comment:
Shouldn't we wait for the removal of the ownership detection code to be merged?

Reply:
Yep, that makes sense.

I will rebase then.

Reply:
I believe next update is rebased already.